GCC 7 issue - name collision with macro.
Need to be evaluated before submitting to upstream.

In file included from /usr/gcc/7/include/c++/7.3.0/bits/std_abs.h:40:0,
                 from /usr/gcc/7/include/c++/7.3.0/cstdlib:77,
                 from /usr/gcc/7/include/c++/7.3.0/stdlib.h:36,
                 from /scratch/userland-gate/components/squid/squid-3.5.27/compat/types.h:29,
                 from /scratch/userland-gate/components/squid/squid-3.5.27/compat/compat.h:70,
                 from /scratch/userland-gate/components/squid/squid-3.5.27/include/squid.h:43,
                 from /scratch/userland-gate/components/squid/squid-3.5.27/helpers/basic_auth/SMB_LM/msntauth.cc:40:
/scratch/userland-gate/components/squid/squid-3.5.27/helpers/basic_auth/SMB_LM/valid.h:18:94: error: expected ',' or '...' before numeric constant
 int Valid_User(char *USERNAME, char *PASSWORD, const char *SERVER, char *BACKUP, const char *DOMAIN);

--- squid-3.5.27/helpers/basic_auth/SMB_LM/valid.h
+++ squid-3.5.27/helpers/basic_auth/SMB_LM/valid.h
@@ -15,7 +15,7 @@
 #define NTV_PROTOCOL_ERROR 2
 #define NTV_LOGON_ERROR 3
 
-int Valid_User(char *USERNAME, char *PASSWORD, const char *SERVER, char *BACKUP, const char *DOMAIN);
+int Valid_User(char *USERNAME, char *PASSWORD, const char *SERVER, char *BACKUP, const char *Domain);
 
 #endif
 
--- squid-3.5.27/helpers/basic_auth/SMB_LM/valid.cc
+++ squid-3.5.27/helpers/basic_auth/SMB_LM/valid.cc
@@ -21,7 +21,7 @@
 
 // BACKUP is unused
 int
-Valid_User(char *USERNAME, char *PASSWORD, const char *SERVER, char *BACKUP, const char *DOMAIN)
+Valid_User(char *USERNAME, char *PASSWORD, const char *SERVER, char *BACKUP, const char *Domain)
 {
     const char *supportedDialects[] = {"PC NETWORK PROGRAM 1.0",
                                        "MICROSOFT NETWORKS 1.03",
@@ -36,7 +36,7 @@
     SMB_Handle_Type con;
 
     SMB_Init();
-    con = SMB_Connect_Server(NULL, SERVER, DOMAIN);
+    con = SMB_Connect_Server(NULL, SERVER, Domain);
     if (con == NULL) {
         return (NTV_SERVER_ERROR);
     }
